src/libmongoc/tests/test-mongoc-exhaust.c

```diff
@@ -28,28 +28,24 @@ skip_if_mongos (void)
 }
 
 
-static int64_t
-get_timestamp (mongoc_client_t *client, mongoc_cursor_t *cursor)
+static uint32_t
+get_generation (mongoc_client_t *client, mongoc_cursor_t *cursor)
 {
    uint32_t server_id;
+   uint32_t generation;
+   mongoc_server_description_t *sd;
+   bson_error_t error;
 
    server_id = mongoc_cursor_get_hint (cursor);
 
-   if (client->topology->single_threaded) {
-      mongoc_topology_scanner_node_t *scanner_node;
-
-      scanner_node = mongoc_topology_scanner_get_node (
-         client->topology->scanner, server_id);
-
-      return scanner_node->timestamp;
-   } else {
-      mongoc_cluster_node_t *cluster_node;
-
-      cluster_node = (mongoc_cluster_node_t *) mongoc_set_get (
-         client->cluster.nodes, server_id);
+   bson_mutex_lock (&client->topology->mutex);
+   sd = mongoc_topology_description_server_by_id (
+      &client->topology->description, server_id, &error);
+   ASSERT_OR_PRINT (sd, error);
+   generation = sd->generation;
+   bson_mutex_unlock (&client->topology->mutex);
```

Comment:
Like my earlier questions, should `mongoc_topology_description_server_by_id` be moved directly after `mongoc_topology_description_server_by_id`? I suppose if the assertion fails we don't care about leaving the mutex locked either way.